src/serialmonitor/serialportctrl.ts

```diff
@@ -50,66 +47,37 @@ export class SerialPortCtrl {
 
   public open(): Promise<any> {
     this._outputChannel.appendLine(`[Starting] Opening the serial port - ${this._currentPort}`);
-    return new Promise((resolve, reject) => {
-      if (this._currentSerialPort && this._currentSerialPort.isOpen) {
-        this._currentSerialPort.close((err) => {
-          if (err) {
-            return reject(err);
-          }
-          this._currentSerialPort = null;
-          return this.open().then(() => {
-            resolve();
-          }, (error) => {
-            reject(error);
-          });
-        });
-      } else {
-        this._currentSerialPort = new SerialPortCtrl.serialport(this._currentPort, { baudRate: this._currentBaudRate, hupcl: false });
-        this._outputChannel.show();
-        this._currentSerialPort.on("open", () => {
-          if (VscodeSettings.getInstance().disableTestingOpen) {
-            this._outputChannel.appendLine("[Warning] Auto checking serial port open is disabled");
-            return resolve();
-          }
-
-          this._currentSerialPort.write("TestingOpen" + "\r\n", (err) => {
-            // TODO: Fix this on the serial port lib: https://github.com/EmergingTechnologyAdvisors/node-serialport/issues/795
-            if (err && !(err.message.indexOf("Writing to COM port (GetOverlappedResult): Unknown error code 121") >= 0)) {
-              this._outputChannel.appendLine(`[Error] Failed to open the serial port - ${this._currentPort}`);
-              reject(err);
-            } else {
-              this._outputChannel.appendLine(`[Info] Opened the serial port - ${this._currentPort}`);
-              this._currentSerialPort.set(["dtr=true", "rts=true"], (err) => {
-                if (err) {
-                  reject(err);
-                }
-              });
-              resolve();
-            }
-          });
-        });
+    this._outputChannel.show();
+
+    if (this._child) {
+        this._child.stdin.write("close\n");
```

Comment:
I think this message needs to be wrapped in JSON. I might suggest just wrapping this logic into a `closeChildProcess` method and reuse that, since I see this same pattern in a few places in this class.

```suggestion
        this._child.stdin.write('{"cmd": "close"}\n');
```